### ISSUE TITLE
LDAP users not getting reflected in AWX 19.3.0

### ISSUE THREAD

**Issue by satbirsp** [ansible/awx#10883]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](http://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I am not entitled to status updates or other assurances.

### Summary

Hi Team 

After doing update/new install for 19.3.0 , Once LDAP new user login it will not show LDAP in front 
![awx](https://user-images.githubusercontent.com/50193939/129516092-eee9cc89-77dd-44af-9543-b263d6b446e7.png)

### AWX version

19.3.0

### Installation method

minikube

### Modifications

no

### Operating system

Ubuntu 20.04

### Web browser

Chrome

### Steps to reproduce

Install/update version 19.3.0 and login via new LDAP user who never log in to AWX . Change LDAP users to SystemAdmin group now new LDAP user will not able to login 

### Expected results

New LDAP user will able to login after changing to systemadmin 

### Actual results

New LDAP user not able to login after changing to systemadmin

**Comment by lumarel** [comment]
I also ran in the exact same problem on a kubernetes cluster.
This is how the login looks in the log:
```
2021-08-17 08:24:57,426 DEBUG    [17b6e01d0c72477e88772d0b15d412c3] awx.analytics.performance request: <WSGIRequest: GET '/api/login/'>, response_time: 0.036s
[pid: 58|app: 0|req: 329/706] <ip-address> () {60 vars in 2482 bytes} [Tue Aug 17 08:24:57 2021] GET /api/login/ => generated 5710 bytes in 37 msecs (HTTP/1.1 200) 10 headers in 483 bytes (1 switches on core 0)
<ip-address> - - [17/Aug/2021:08:24:57 +0000] "GET /api/login/ HTTP/1.1" 200 5710 "https://<fqdn>/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:78.0) Gecko/20100101 Firefox/78.0" "<ip-address>"
2021-08-17 08:24:57,514 WARNING  [13b719369dce43eaa20cc5e32bdbffc3] awx.api.generics Login failed for user <ldap-username> from <ip-address>
2021-08-17 08:24:57,519 DEBUG    [13b719369dce43eaa20cc5e32bdbffc3] awx.analytics.performance request: <WSGIRequest: POST '/api/login/'>, response_time: 0.066s
2021-08-17 08:24:57,519 WARNING  [13b719369dce43eaa20cc5e32bdbffc3] django.request Unauthorized: /api/login/
2021-08-17 08:24:57,519 WARNING  [13b719369dce43eaa20cc5e32bdbffc3] django.request Unauthorized: /api/login/
[pid: 60|app: 0|req: 104/707] <ip-address> () {66 vars in 2640 bytes} [Tue Aug 17 08:24:57 2021] POST /api/login/ => generated 5924 bytes in 67 msecs (HTTP/1.1 401) 10 headers in 493 bytes (1 switches on core 0)
<ip-address> - - [17/Aug/2021:08:24:57 +0000] "POST /api/login/ HTTP/1.1" 401 5924 "https://<fqdn>/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:78.0) Gecko/20100101 Firefox/78.0" "<ip-address>"
```

To me it looks like the useraccount does not get created as an LDAP user, because if the password gets changed on the first login the login also works on the second time (off course with the local login not against LDAP)

**Comment by docholiday** [comment]
I can confirm the problem (also on a minikube installation) and I share the impression, that the useraccounts seem to be created as local accounts instead of ldap ones.

**Comment by satbirsp** [comment]
Yes its on minikube . Initially users will able to login using there password . Once you change role from normal to System Admin they will not able to login , Yes i cannot see LDAP as sufix once user login

**Comment by TristisOris** [comment]
same problem. old accounts from 19.2 works, new - not "wrong password"

**Comment by ZachHoiberg** [comment]
Same problem here. Deleting the user allows another successful sign-on, but fails after that initial sign-on. LDAP suffix is also absent.

Additionally, I was able to work around this by downgrading to 19.2.0, signing in with the user, and then upgrading to 19.3.0 again. They will be properly tagged with LDAP and should be able to sign-in as normal.

**Comment by kurokobo** [comment]
@shanemcd @tchellomello 
I faced the same issue, so I digged into the code. 
This issue seems to be caused by PR ansible/awx#9703.

Even if the user login with a new LDAP account, [the `force_user_update` flag added by that PR never be `True`](https://github.com/ansible/awx/blob/fd3475a813e84789f47213c7ce8bf869995de1fa/awx/sso/backends.py#L367-L369), so [the followed `user.save()` and `profile.save()`](https://github.com/ansible/awx/blob/fd3475a813e84789f47213c7ce8bf869995de1fa/awx/sso/backends.py#L399-L404) are never invoked for that new user.
Therefore `ldap_dn` is not updated, the user is treated as just a local user, not as an external user.

I'm not familiar with the issue on which the PR is based, but I think we need to at least make sure that `*.save()` is executed for new users.

**Comment by tchellomello** [comment]
@kurokobo  thanks for the heads up on this. Yes, I tested here and the issue is because the `ldap_dn` is not being added. My apologies for that. Yet, to avoid the double LDAP updates, we can call the `user.save()` only when required. When the user does not exist yet in the database, trying to access the `user.profile` throws a `ValueError` exception:

```python
>>> profile = user.profile
*** ValueError: save() prohibited to prevent data loss due to unsaved related object 'user'.
>>> User.objects.filter(username=user.username).count()
0
``` 

Working on some tests.  Something like this:

```python
    # To avoid double LDAP updates, force user.save if ldap_dn does not exist
    try:
        profile = user.profile
    except ValueError:
        force_user_update = True

    if force_user_update:
        user.save()

    # Update user profile to store LDAP DN.
    profile = user.profile
    if profile.ldap_dn != ldap_user.dn:
        profile.ldap_dn = ldap_user.dn
        profile.save()

```

This seems to be working on my lab here

![image](https://user-images.githubusercontent.com/809840/131267068-685f6102-9cff-4407-8a64-603932bb7cf3.png)

**Comment by shanemcd** [comment]
This should be fixed with AWX 19.4.0

**Comment by lumarel** [comment]
I can confirm, it works again, thank you! 👌🏻

**Comment by ktibi** [comment]
Hello,

Not fix on my AWX 19.4.0.

Ldap User can login only one time.

**Comment by tiagodread** [comment]
This is now fixed

![image](https://user-images.githubusercontent.com/12464858/139093110-2e1dc87f-ff28-47cf-8b2a-f3c04c706dec.png)
